core/src/main/scala/it/scalachess/core/pieces/Piece.scala

```diff
@@ -1,34 +1,39 @@
 package it.scalachess.core.pieces
 
 import it.scalachess.core.board.Position
-import it.scalachess.core.colors.Color
+import it.scalachess.core.colors.{ Black, Color, White }
 
 final case class Piece(color: Color, pieceType: PieceType) {
 
   def canAttack(start: Position, end: Position): Boolean = {
-    val rowD: Int = start rowDistance end
-    val colD: Int = start colDistance end
+    val colDistanceInt = start colDistanceAbs end
     pieceType match {
       case King   => start isAdjacentTo end
-      case Queen  => (start isAdjacentTo end) || (start isStraightTo end)
+      case Queen  => (start isDiagonalTo end) || (start isStraightTo end)
       case Rook   => start isStraightTo end
       case Bishop => start isDiagonalTo end
-      case Knight => (rowD == 2 && colD == 1) || (rowD == 1 && colD == 2)
-      case Pawn   => Piece.canPawnAttack(start, end, color)
+      case Knight =>
+        val rowDistanceInt = start rowDistanceAbs end
+        (rowDistanceInt == 2 && colDistanceInt == 1
+        || rowDistanceInt == 1 && colDistanceInt == 2)
+      case Pawn =>
+        val rowDistanceInt = end rowDistanceInt start
```

Comment:
no need to specify the type in the variable name
`rowDistance: Int = end rowDistanceInt start`